design/gateway-apis-implementation.md

```diff
@@ -1,204 +1,420 @@
+
+Contour’s implementation of the Operator has taken a similar approach where based upon the configuration specified in the Envoy Endpoint section, the operator would translate that configuration into a set of annotations on the Envoy service.
+
+Initially Contour can do the same work to move that same logic and expose Envoy via annotations on the Service Type=LoadBalancer.
+
+The future looking way to approach this is to utilize GatewayAPI fully and instead of exposing a service with type=LoadBalancer, Contour can create a GatewayClass, Gateway which selects a TCPRoute used to send traffic to the Envoy fleet.
+The details defined in the Contour configuration CRD will be used to pre-populate annotations that will be stored in a Kubernetes configmap and referenced from the GatewayClass.Spec.ParametersRef.
+The Gateway Stub controller will be utilized to configure a new Kubernetes Service with the annotations defined in the configmap on the GatewayClass and send traffic to Envoy.
+Once implementers catch up, specific implementations will be utilized and the stub controller can be deprecated.
+
+New fields will be added to the Contour configuration file which will allow users to define the GVK of a resource used in the GatewayClass along with the ControllerName to be used.
+The Operator will still auto-create the GatewayClass/Gateway which will allow the operator to still maintain the proper listeners matching the Gateway Contour is processing, but allow for custom implementations without the need for code changes to Contour.
+
+![alt_text](images/gatewayapi-external-gateway.jpg)
+
+### WorkFlows
+
+There are many different ways that Contour can be deployed depending on user needs when interacting with the Contour Operator.
+
+**Assertion**:_ If users are deploying Contour with the Operator, then they are looking for a fully managed environment, meaning, instances of Contour are managed by the Operator, and instances of Envoy are managed by Contour._
+
+When users want to deploy Contour without the Operator, then they have the ability to choose if Envoy should be managed dynamically by Contour or not.
+
+**Assertion**:_ A Contour Configuration CRD is required for Contour to function in all scenarios._
+
+**Assertion**:_ The GatewayClass.Spec.ParamsRef will not be defined since all the work to define how a Contour/Envoy is configured is defined in the Contour configuration CRD._
+
+#### Operator with GatewayAPI (Envoy is managed by convention)
+
+In this mode, the Operator is managing instances of Contour. Each instance of Contour manages its own Envoy fleet by convention (since the Operator is in use). Contour will expose the Envoy fleet by managing a Gateway/GatewayClass/TCPRoute to get traffic to the Envoy instances. The Contour configuration CRD will define how that traffic will get routed or if another Gateway should be managing the traffic (i.e. custom).
+
+1. Deploy Operator to Cluster
```

Comment:
nit: I think some of the markdown rendering on the lists in this and following sections might need a look just b/c of indenting etc. to make sure everything is how you expect it

Reply:
Yeah I'll clean that up. I used a plugin to move from gDocs to markdown and it didn't do a great job in some places.